tokio/tests/io_async_fd.rs

```diff
@@ -500,10 +500,10 @@ fn driver_shutdown_wakes_currently_pending() {
 
     std::mem::drop(rt);
 
-    // Being awoken by a rt drop does not return an error, currently...
-    let _ = futures::executor::block_on(readable).unwrap();
+    // The future was initialized **before** dropping the rt
+    assert_err!(futures::executor::block_on(readable));
```

Comment:
@bdonlan wdyt?

Reply:
This is fine, I think - I viewed it as spurious wakeups being allowable so the test tolerated this one as long as we eventually got to an `Err`.